Event streaming page (#6444)
Particular/docs.particular.net: nservicebus/event-streaming/index.md

---
title: Event stream processing

> summary: Combining the Particular Service Platform with an event stream processor (Apache Kafka, Azure Event Hubs, or Amazon Kinesis)

Reviewer: Should we use the term "integration" instead?

> summary: Combining the Particular Service Platform with an event stream processor (Apache Kafka, Azure Event Hubs, or Amazon Kinesis)

Author: ```suggestion
summary: Integrating the Particular Service Platform with an event stream processor (Apache Kafka, Azure Event Hubs, or Amazon Kinesis)
```

callsToAction:
  - interest-apache-kafka
  - interest-azure-event-hubs
  - interest-amazon-kinesis
reviewed: 2024-02-06
related:
  - samples/azure-functions/service-bus-kafka
---

> 

Reviewer 2: It would be nice and probably more engaging to have a nice picture of Kafka.
In my previous attempt at this, I went with:
![Kafka logo](https://apache.org/logos/res/kafka/kafka_highres.png)

Author: I took it out because it looked odd, especially with the callouts. I tried moving it around to make it fit right and it causes a bunch of misalignment.

> _Event stream processing_ is a practice of taking action on a series of datapoints that originate from a system that continuously creates data.

Reviewer: ```suggestion
_Event stream processing_ is an architectural style in which a continuous stream of events serve as the basis to take further action in a system.
```

There are great tools that facilitate event streaming including:

- [Apache Kafka](https://kafka.apache.org/)
- [Azure Event Hubs](https://azure.microsoft.com/en-us/services/event-hubs/)
- [Amazon Kinesis](https://aws.amazon.com/kinesis/).

> Although these tools and the Particular Service Platform both use the term _event_, [an event stream is not a queue](https://particular.net/blog/lets-talk-about-kafka).

Reviewer 2: I am unsure if we should have a link to somewhere else so early. I guess I am concerned the user may click on that and move on.
Could links like this be added as "Other resources" at the end of the page?

Author: I was also concerned by this. Also about biasing the type of feedback people might give. If we state an strong opinion we preclude someone who thinks they really want a Kafka transport.

> Event streaming tools are great for situations where a large amount of data needs to be processed looking for patterns or changes in state. Individual events are not important on their own, and only by processing many of them can interesting events be extracted.

Reviewer: ```suggestion
Event streaming tools are great for situations where a large amount of data needs to be processed, looking for patterns or changes in state. Individual events are not meaningful, but meaningful insights can be extracted by processing many.
```

We could also maybe use terminology like a singular event (unmeaningful) and business event (based on multiple events leading to business insight).

> When something interesting is detected, that is a business event, and _it is time to use a message queue_. Message queues excel at handling important events, where each individual event is important. Business events should be successfully processed once, and only once, and should be retried until they are successfully processed.

Reviewer: ```suggestion
When something interesting is detected, that is a business event, and _it is time to use a message queue_. Message queues excel at handling business events, where each individual event is meaningful and important. Business events should be successfully processed once, and only once, and should be retried until they are successfully processed.
```

> To facilitate this, message stream processors that detect interesting patterns can turn them into NServiceBus messages and events.

Reviewer 2: I wonder if this section adds much value. 
To me, it feels solution-oriented, which may void getting good feedback from the users.
